hack/verify-checkoutput.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script does a fast type check of kubernetes code for all platforms.
+# Usage: `hack/verify-typecheck.sh`.
```

Comment:
This description is consistent with my expectation: without parameters, the script will check *all* binaries for which checking is enabled. However, the implementation doesn't match that behavior (yet).

To implement it, we need to define what "all binaries" and "checking is enabled" means. I think the answers are:
- find all directories with a "main" package
- there are one or more .md files inside that directory which contain test cases

Reply:
Ok couple of questions. Do we expect a coupling between the .md file and the main package ? If so, we need to validate that the test case corresponds to package main. How about we initially only target README.md and main.go ? I see a number of such instances that would be caught by this restriction [1].

[1] 
staging/src/k8s.io/component-base/logs/kube-log-runner
 staging/src/k8s.io/client-go/examples/dynamic-create-update-delete-deployment/

Reply:
You mean that you want to look for a `main.go` file instead of `main` package? As you noticed, Go does not require that such a file exists. I think it is better to to look for `^package main$` in .go files, which should be reasonably fast and pretty accurate.

I don't know whether a package may be called `main` without implementing a `main` function, i.e. not implement a command. Even if it is allowed, it would be so confusing that no-one is doing it.

Reply:
I have updated the code and the README.md to reflect usage of main package and the markdown file.